core/server/controllers/frontend/index.js

```diff
@@ -67,7 +67,10 @@ frontendControllers = {
             }
 
             // CASE: we only support /:slug format for pages
-            if (post.page && post.url !== req.path) {
+            // However, given that a url with `/edit/` appended is different
+            // than the post url, we must remove `/edit/` from the url if it
+            // is an edit url.
+            if (post.page && post.url !== ((lookup.isEditURL) ? req.path.replace(/edit\/$/, '') : req.path)) {
```

Comment:
I wonder why we can't switch the calling order?
`lookup.isEditURL` with `post.page && post.url !== req.path`. 
That should work in my opinion?